atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -381,10 +381,11 @@ public void disableReadWriteConflictChecking(TableReference tableRef) {
         if (Iterables.isEmpty(rows)) {
```

Comment:
Iterables.isEmpty might create an iterator already. So I believe you need to create a copy before this if statement.

Reply:
You're right - it does. So if you give an inconsistent iterable we actually rely on the ordering we get the *second* time we get an iterator. We're saying the ordering doesn't matter in that case anyway, so this is fine for the guarantees we've given.

Reply:
Cool